src/pages/Main/Main.js

```diff
@@ -0,0 +1,434 @@
+      kindOfcolor === 0
+    ) {
+      console.log(
+        makeFirstFilterQuery(collectionFilter).concat(
+          makeNthFilterQuery(sizeFilter)
+        )
+      );
+    } else if (
+      kindOfBrand === 0 &&
+      kindOfSize !== 0 &&
+      kindOfcollection === 0 &&
+      kindOfcolor !== 0
+    ) {
+      console.log(
+        makeFirstFilterQuery(colorFilter).concat(makeNthFilterQuery(sizeFilter))
+      );
+    } else if (
+      kindOfBrand === 0 &&
+      kindOfSize === 0 &&
+      kindOfcollection !== 0 &&
+      kindOfcolor !== 0
+    ) {
+      console.log(
+        makeFirstFilterQuery(collectionFilter).concat(
+          makeNthFilterQuery(colorFilter)
+        )
+      );
+    } else if (
+      kindOfBrand !== 0 &&
+      kindOfSize !== 0 &&
+      kindOfcollection !== 0 &&
+      kindOfcolor === 0
+    ) {
+      console.log(
+        makeFirstFilterQuery(brandFilter)
+          .concat(makeNthFilterQuery(collectionFilter))
+          .concat(makeNthFilterQuery(sizeFilter))
+      );
+    } else if (
+      kindOfBrand !== 0 &&
+      kindOfSize !== 0 &&
+      kindOfcollection === 0 &&
+      kindOfcolor !== 0
+    ) {
+      console.log(
+        makeFirstFilterQuery(brandFilter)
+          .concat(makeNthFilterQuery(colorFilter))
+          .concat(makeNthFilterQuery(sizeFilter))
+      );
+    } else if (
+      kindOfBrand !== 0 &&
+      kindOfSize === 0 &&
+      kindOfcollection !== 0 &&
+      kindOfcolor !== 0
+    ) {
+      console.log(
+        makeFirstFilterQuery(brandFilter)
+          .concat(makeNthFilterQuery(collectionFilter))
+          .concat(makeNthFilterQuery(colorFilter))
+      );
+    } else if (
+      kindOfBrand === 0 &&
+      kindOfSize !== 0 &&
+      kindOfcollection !== 0 &&
+      kindOfcolor !== 0
+    ) {
+      console.log(
+        makeFirstFilterQuery(collectionFilter)
+          .concat(makeNthFilterQuery(colorFilter))
+          .concat(makeNthFilterQuery(sizeFilter))
+      );
+    } else if (
+      kindOfBrand !== 0 &&
+      kindOfSize !== 0 &&
+      kindOfcollection !== 0 &&
+      kindOfcolor !== 0
+    ) {
+      console.log(
+        makeFirstFilterQuery(brandFilter)
+          .concat(makeNthFilterQuery(collectionFilter))
+          .concat(makeNthFilterQuery(colorFilter))
+          .concat(makeNthFilterQuery(sizeFilter))
+      );
+    }
+  };
+
+  useEffect(() => {
+    fetch('/Data/Products.json')
+      .then(result => result.json())
+      .then(ProductListData => {
+        setproductList(ProductListData.Products);
+      });
+  }, []);
+
+  useEffect(() => {
+    fetchByFilterQuery();
+  }, [brandFilter, collectionFilter, colorFilter, sizeFilter]);
+
+  const brandFilterControl = e => {
+    if (brandFilter.brand.includes(e.target.name)) {
+      setBrandFilter({
+        brand: brandFilter.brand.filter(el => el !== e.target.name),
+      });
+    } else {
+      setBrandFilter({ brand: [...brandFilter.brand, e.target.name] });
+    }
+  };
```

Comment:
onchange 함수도 상태 값이 변경됨에 따라 바꿔줄 수 있겠네요!

Reply:
```suggestion
const filterControl = (e, filterTypeQuery) => {
    if (productfilters[filterTypeQuery].includes(e.target.name)) {
      setProductFilters(prev => {
        return {
          ...prev,
          [filterTypeQuery]: productfilters[filterTypeQuery].filter(
            el => el !== e.target.name
          ),
        };
      });
    } else {
      setProductFilters(prev => {
        return {
          ...prev,
          [filterTypeQuery]: [
            ...productfilters[filterTypeQuery],
            e.target.name,
          ],
        };
      });
    }
  };
```

네 기존의 필터 state 별로 있던 함수 4개를 위와 같이 하나로 합쳤습니다